Explorer/Assets/DCL/Character/CharacterMotion/Components/PlayerTeleportIntent.cs

```diff
@@ -33,17 +34,25 @@ public JustTeleported(int expireFrame, Vector2Int parcel)
         /// </summary>
         public readonly AsyncLoadProcessReport? AssetsResolution;
 
-        public readonly float CreationTime;
+        public bool TimedOut => Time.realtimeSinceStartup - creationTime > TIMEOUT.TotalSeconds;
 
-        public PlayerTeleportIntent(Vector3 position, Vector2Int parcel, CancellationToken cancellationToken, AsyncLoadProcessReport? assetsResolution = null)
+        public PlayerTeleportIntent(SceneEntityDefinition? sceneDef, Vector2Int parcel, CancellationToken cancellationToken, AsyncLoadProcessReport? assetsResolution = null)
         {
-            Position = position;
             Parcel = parcel;
             CancellationToken = cancellationToken;
             AssetsResolution = assetsResolution;
-            CreationTime = Time.realtimeSinceStartup;
+            creationTime = Time.realtimeSinceStartup;
+            SceneDef = sceneDef;
         }
+    }
+
+    public readonly struct TeleportPosition
```

Comment:
I dont think you need this struct. 

As far as I see, you can add the Vector3 position to `PlayerTeleportIntent` and avoid the complexity of handling an extra component.

Reply:
Initially I had it in the `PlayerTeleportIntent`, yes. However, `GlobalWorldActions.cs` was making like "force" teleportation. The only solution there was to have Position as nullable `Vector3?` in `PlayerTeleportIntent` and have null-check in the `TeleportPositionCalculationSystem.cs` to skip calculation if it is not null. So I decided to go with the classical ECS approach where components are split by atomic data (but I only did it for position in this case)

Reply:
removed